File: android/src/main/kotlin/com/apadmi/flutter_security_suite/FlutterSecuritySuitePlugin.kt
package com.apadmi.flutter_security_suite

import SecuritySuiteApi
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

class FlutterSecuritySuitePlugin: FlutterPlugin {
  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    val securityApi = ServiceLocator.getSecuritySuite(flutterPluginBinding.applicationContext)
    SecuritySuiteApi.setUp(flutterPluginBinding.binaryMessenger, securityApi)
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    // Intentionally blank.
  }
}

File: ios/flutter_security_suite/Sources/flutter_security_suite/FlutterSecuritySuitePlugin.swift
import Flutter
import UIKit

public class FlutterSecuritySuitePlugin: NSObject, FlutterPlugin {
    
    let securitySuite: SecuritySuiteApi
    
    override init() {
        self.securitySuite = ServiceLocator.getSecuritySuiteApi()
    }
    
    public static func register(with registrar: FlutterPluginRegistrar) {
        let plugin = FlutterSecuritySuitePlugin()
        let securitySuite = plugin.securitySuite
        SecuritySuiteApiSetup.setUp(binaryMessenger: registrar.messenger(), api: securitySuite)
        registrar.publish(plugin)
    }
}
